### PR TITLE
[Azure Spring Cloud][spring-cloud] Migrate to track2 SDK

### PR DESCRIPTION
**Pull request by ShichaoQiu**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
spring-cloud

**Review by jsntcy — APPROVED** [review]

File: src/spring-cloud/azext_spring_cloud/custom.py [hunk 1]
```diff
@@ -141,20 +147,10 @@ def spring_cloud_update(cmd, client, resource_group, name, app_insights_key=None
...
-            sdk_no_wait(no_wait, client.monitoring_settings.update_put,
-                        resource_group_name=resource_group, service_name=name, properties=trace_properties)
-
-    # update service tags
-    if tags is not None:
-        updated_resource.tags = tags
-        update_service_tags = True
-
-    if update_service_tags is False and update_service_sku is False:
-        return resource
```
Comment (zhoxing-ms) [inline src/spring-cloud/azext_spring_cloud/custom.py:153]: Why is this logic no longer needed?  Will deleting this logic cause breaking change to users?

**Comment by zhoxing-ms** [comment]
There are no any recorded yaml files here. May I ask have you rerun the relevant tests?

File: src/spring-cloud/azext_spring_cloud/custom.py (hunk 1 above)
Comment (ShichaoQiu) [reply src/spring-cloud/azext_spring_cloud/custom.py:153]: Good catch, actually, they are the duplicate code

Comment (zhoxing-ms) [reply src/spring-cloud/azext_spring_cloud/custom.py:153]: OK

**Review by zhoxing-ms — APPROVED** [review]

**Comment by ShichaoQiu** [comment]
LGTM

**Pull request merged**